File: sdk/go/domains.go
package qeetid

import (
	"context"
	"net/http"
	"net/url"
)

type Domain struct {
	ID                string `json:"id"`
	TenantID          string `json:"tenant_id"`
	Domain            string `json:"domain"`
	Verified          bool   `json:"verified"`
	VerificationToken string `json:"verification_token,omitempty"`
	CreatedAt         string `json:"created_at"`
	UpdatedAt         string `json:"updated_at,omitempty"`
}

type CreateDomainInput struct {
	Domain string `json:"domain"`
}

type Domains struct{ http *httpClient }

func (r *Domains) Create(ctx context.Context, tenantID string, in CreateDomainInput) (*Domain, error) {
	var out Domain
	err := r.http.do(ctx, http.MethodPost, "/v1/tenants/"+url.PathEscape(tenantID)+"/domains", nil, in, &out, false)
	return &out, err
}

func (r *Domains) Get(ctx context.Context, tenantID, id string) (*Domain, error) {
	var out Domain
	err := r.http.do(ctx, http.MethodGet, "/v1/tenants/"+url.PathEscape(tenantID)+"/domains/"+url.PathEscape(id), nil, nil, &out, true)
	return &out, err
}

func (r *Domains) Delete(ctx context.Context, tenantID, id string) error {
	return r.http.do(ctx, http.MethodDelete, "/v1/tenants/"+url.PathEscape(tenantID)+"/domains/"+url.PathEscape(id), nil, nil, nil, true)
}

func (r *Domains) Verify(ctx context.Context, tenantID, id string) (*Domain, error) {
	var out Domain
	err := r.http.do(ctx, http.MethodPost, "/v1/tenants/"+url.PathEscape(tenantID)+"/domains/"+url.PathEscape(id)+"/verify", nil, struct{}{}, &out, false)
	return &out, err
}

func (r *Domains) List(ctx context.Context, tenantID string) ([]Domain, error) {
	var env struct {
		Items []Domain `json:"items"`
		Data  []Domain `json:"data"`
	}
	if err := r.http.do(ctx, http.MethodGet, "/v1/tenants/"+url.PathEscape(tenantID)+"/domains", nil, nil, &env, true); err != nil {
		return nil, err
	}
	if env.Items != nil {
		return env.Items, nil
	}
	return env.Data, nil
}

File: sdk/js/node/src/domains.ts
import type { HttpClient } from "./client.js";

export interface Domain {
  id: string;
  tenant_id: string;
  domain: string;
  verified: boolean;
  verification_token?: string;
  created_at: string;
  updated_at?: string;
}

export interface CreateDomainInput {
  domain: string;
}

interface ListEnvelope<T> {
  items?: T[];
  data?: T[];
}

export class Domains {
  constructor(private readonly http: HttpClient) {}

  create(tenantId: string, input: CreateDomainInput): Promise<Domain> {
    return this.http.post<Domain>(`/v1/tenants/${encodeURIComponent(tenantId)}/domains`, input);
  }

  get(tenantId: string, id: string): Promise<Domain> {
    return this.http.get<Domain>(
      `/v1/tenants/${encodeURIComponent(tenantId)}/domains/${encodeURIComponent(id)}`,
    );
  }

  delete(tenantId: string, id: string): Promise<void> {
    return this.http.delete<void>(
      `/v1/tenants/${encodeURIComponent(tenantId)}/domains/${encodeURIComponent(id)}`,
    );
  }

  verify(tenantId: string, id: string): Promise<Domain> {
    return this.http.post<Domain>(
      `/v1/tenants/${encodeURIComponent(tenantId)}/domains/${encodeURIComponent(id)}/verify`,
      {},
    );
  }

  async list(tenantId: string): Promise<Domain[]> {
    const res = await this.http.get<ListEnvelope<Domain>>(
      `/v1/tenants/${encodeURIComponent(tenantId)}/domains`,
    );
    return res.items ?? res.data ?? [];
  }
}
